ocaml/testsuite/tests/templates/basic/bad_instance_arg_value_wrong_type.reference

```diff
@@ -0,0 +1,3 @@
+File "bad_instance_arg_value_wrong_type.ml", line 1:
+Error: The module "List_monoid" was expected to satisfy the parameter "Category"
+       but "list_monoid.cmi" was compiled with -as-argument-for "Monoid".
```

Comment:
I feel we should avoid talking at different levels for "actual" and "expected"? I.e., either we only talk about the high-level "satisfy parameter", or we only talk about the low-level "compiled with -as-argument-for Monoid". How about:

> Error: The module "List_monoid" was expected to satisfy the parameter "Category", but was compiled to satisfy "Monoid".